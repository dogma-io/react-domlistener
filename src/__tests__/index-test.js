import DOMListener from '../index'
import {mount} from 'enzyme'
import React from 'react'

const TESTS = [
  {
    desc: 'when not capture, once, or passive',
    props: {},
  },
  {
    desc: 'when capture',
    props: {capture: true},
  },
  {
    desc: 'when once',
    props: {once: true},
  },
  {
    desc: 'when passive',
    props: {passive: true},
  },
  {
    desc: 'when capture and once',
    props: {capture: true, once: true},
  },
  {
    desc: 'when capture and passive',
    props: {capture: true, passive: true},
  },
  {
    desc: 'when once and passive',
    props: {once: true, passive: true},
  },
  {
    desc: 'when capture, once, and passive',
    props: {capture: true, once: true, passive: true},
  },
]

describe('DOMListener', () => {
  TESTS.forEach(({desc, props}) => {
    describe(desc, () => {
      let listener, target, wrapper

      beforeEach(() => {
        listener = jest.fn()
        target = {
          addEventListener: jest.fn(),
          removeEventListener: jest.fn(),
        }
        wrapper = mount(
          <DOMListener
            listener={listener}
            target={target}
            type="click"
            {...props}
          />,
        )
      })

      it('should function as expected when mounted', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('should function as expected when unmounted', () => {
        wrapper.unmount()
        expect({target, wrapper}).toMatchSnapshot()
      })
    })
  })
})
