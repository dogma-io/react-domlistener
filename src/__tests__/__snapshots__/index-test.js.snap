// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOMListener when capture and once should function as expected when mounted 1`] = `
<DOMListener
  capture={true}
  listener={[MockFunction]}
  once={true}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            Object {
              "capture": true,
              "once": true,
              "passive": undefined,
            },
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when capture and once should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": true,
            "once": true,
            "passive": undefined,
          },
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": true,
            "once": true,
            "passive": undefined,
          },
        ],
      ],
    },
  },
  "wrapper": "",
}
`;

exports[`DOMListener when capture and passive should function as expected when mounted 1`] = `
<DOMListener
  capture={true}
  listener={[MockFunction]}
  passive={true}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            Object {
              "capture": true,
              "once": undefined,
              "passive": true,
            },
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when capture and passive should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": true,
            "once": undefined,
            "passive": true,
          },
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": true,
            "once": undefined,
            "passive": true,
          },
        ],
      ],
    },
  },
  "wrapper": "",
}
`;

exports[`DOMListener when capture should function as expected when mounted 1`] = `
<DOMListener
  capture={true}
  listener={[MockFunction]}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            true,
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when capture should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          true,
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          true,
        ],
      ],
    },
  },
  "wrapper": "",
}
`;

exports[`DOMListener when capture, once, and passive should function as expected when mounted 1`] = `
<DOMListener
  capture={true}
  listener={[MockFunction]}
  once={true}
  passive={true}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            Object {
              "capture": true,
              "once": true,
              "passive": true,
            },
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when capture, once, and passive should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": true,
            "once": true,
            "passive": true,
          },
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": true,
            "once": true,
            "passive": true,
          },
        ],
      ],
    },
  },
  "wrapper": "",
}
`;

exports[`DOMListener when not capture, once, or passive should function as expected when mounted 1`] = `
<DOMListener
  listener={[MockFunction]}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            false,
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when not capture, once, or passive should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          false,
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          false,
        ],
      ],
    },
  },
  "wrapper": "",
}
`;

exports[`DOMListener when once and passive should function as expected when mounted 1`] = `
<DOMListener
  listener={[MockFunction]}
  once={true}
  passive={true}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            Object {
              "capture": undefined,
              "once": true,
              "passive": true,
            },
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when once and passive should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": undefined,
            "once": true,
            "passive": true,
          },
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": undefined,
            "once": true,
            "passive": true,
          },
        ],
      ],
    },
  },
  "wrapper": "",
}
`;

exports[`DOMListener when once should function as expected when mounted 1`] = `
<DOMListener
  listener={[MockFunction]}
  once={true}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            Object {
              "capture": undefined,
              "once": true,
              "passive": undefined,
            },
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when once should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": undefined,
            "once": true,
            "passive": undefined,
          },
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": undefined,
            "once": true,
            "passive": undefined,
          },
        ],
      ],
    },
  },
  "wrapper": "",
}
`;

exports[`DOMListener when passive should function as expected when mounted 1`] = `
<DOMListener
  listener={[MockFunction]}
  passive={true}
  target={
    Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [MockFunction],
            Object {
              "capture": undefined,
              "once": undefined,
              "passive": true,
            },
          ],
        ],
      },
      "removeEventListener": [MockFunction],
    }
  }
  type="click"
/>
`;

exports[`DOMListener when passive should function as expected when unmounted 1`] = `
Object {
  "target": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": undefined,
            "once": undefined,
            "passive": true,
          },
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "click",
          [MockFunction],
          Object {
            "capture": undefined,
            "once": undefined,
            "passive": true,
          },
        ],
      ],
    },
  },
  "wrapper": "",
}
`;
